spacy/pipeline/lemmatizer.py

```diff
@@ -0,0 +1,277 @@
+from typing import Optional, List, Dict
+
+from thinc.api import Model
+
+from .pipe import Pipe
+from ..errors import Errors
+from ..language import Language
+from ..lookups import Lookups, load_lookups
+from ..parts_of_speech import NAMES as UPOS_NAMES
+from ..tokens import Doc, Token
+from ..vocab import Vocab
+from .. import util
+
+
+@Language.factory(
+    "lemmatizer",
+    assigns=["token.lemma"],
+    default_config={
+        "model": None,
+        "mode": "lookup",
+        "lookups": None,
+        "overwrite": False,
+    },
+    scores=["lemma_acc"],
+    default_score_weights={"lemma_acc": 1.0},
+)
+def make_lemmatizer(
+    nlp: Language,
+    model: Optional[Model],
+    name: str,
+    mode: str,
+    lookups: Optional[Lookups],
+    overwrite: bool = False,
+):
+    lookups = Lemmatizer.load_lookups(nlp.lang, mode, lookups)
+    return Lemmatizer(
+        nlp.vocab, model, name, mode=mode, lookups=lookups, overwrite=overwrite
+    )
+
+
+class Lemmatizer(Pipe):
+    """
+    The Lemmatizer supports simple part-of-speech-sensitive suffix rules and
+    lookup tables.
+
+    DOCS: https://spacy.io/api/lemmatizer
+    """
+
+    @classmethod
+    def get_lookups_config(cls, mode: str) -> Dict:
```

Comment:
Is it correct to be more specific on the `Dict` here?

```suggestion
    def get_lookups_config(cls, mode: str) -> Dict[str, List[str]]:
```

Reply:
I was trying to leave it more open-ended. You could implement a lemmatizer with its own set of `get_lookups_config` and `load_lookups` that don't follow this exactly. (I have to say I'm not totally sure how it works out in terms of the typing, though.)